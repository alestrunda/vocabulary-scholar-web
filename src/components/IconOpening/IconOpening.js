import React from 'react';
import PropTypes from 'prop-types';

const IconOpening = ({ children, icon, ...restProps }) => {
  return (
    <div className="icon-opening" {...restProps}>
      <div className="icon-opening__icon">{icon}</div>
      <div className="icon-opening__content">{children}</div>
    </div>
  );
};

IconOpening.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  icon: PropTypes.element,
};

export default IconOpening;
