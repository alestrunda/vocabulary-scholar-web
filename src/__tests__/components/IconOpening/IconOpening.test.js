import React from 'react';
import { shallow } from 'enzyme';

import IconOpening from '../../../components/IconOpening';
import ActionFavorite from 'material-ui/svg-icons/action/favorite';

it('renders icon', () => {
  const iconOpening = shallow(<IconOpening icon={<ActionFavorite />} />);
  expect(iconOpening).toMatchSnapshot();
});

it('renders content', () => {
  const iconOpeningContent = 'Lorem Ipsum';
  const iconOpening = shallow(
    <IconOpening icon={<ActionFavorite />}>{iconOpeningContent}</IconOpening>
  );
  expect(iconOpening).toMatchSnapshot();
});
