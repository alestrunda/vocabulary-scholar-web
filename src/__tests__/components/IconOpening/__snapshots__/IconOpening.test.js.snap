// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconOpening
    icon={<ActionFavorite />}
  >
    Lorem Ipsum
  </IconOpening>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon-opening__icon"
        >
          <ActionFavorite />
        </div>,
        <div
          className="icon-opening__content"
        >
          Lorem Ipsum
        </div>,
      ],
      "className": "icon-opening",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ActionFavorite />,
          "className": "icon-opening__icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem Ipsum",
          "className": "icon-opening__content",
        },
        "ref": null,
        "rendered": "Lorem Ipsum",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon-opening__icon"
          >
            <ActionFavorite />
          </div>,
          <div
            className="icon-opening__content"
          >
            Lorem Ipsum
          </div>,
        ],
        "className": "icon-opening",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ActionFavorite />,
            "className": "icon-opening__icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem Ipsum",
            "className": "icon-opening__content",
          },
          "ref": null,
          "rendered": "Lorem Ipsum",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders icon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconOpening
    icon={<ActionFavorite />}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon-opening__icon"
        >
          <ActionFavorite />
        </div>,
        <div
          className="icon-opening__content"
        />,
      ],
      "className": "icon-opening",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ActionFavorite />,
          "className": "icon-opening__icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "icon-opening__content",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon-opening__icon"
          >
            <ActionFavorite />
          </div>,
          <div
            className="icon-opening__content"
          />,
        ],
        "className": "icon-opening",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ActionFavorite />,
            "className": "icon-opening__icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "icon-opening__content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
